tests/js/server/replication/replication-sync.js

```diff
@@ -215,6 +217,153 @@ function BaseTestConfig () {
       );
     },
 
+    testExistingIndexId: function () {
+      connectToMaster();
+
+      compare(
+        function (state) {
+          const c = db._create(cn);
+          state.indexDef = {type: 'skiplist', id: '1234567', fields: ['value']};
+          c.ensureIndex(state.indexDef);
+          state.indexProps = c.index(state.indexDef.id);
+        },
+        function (state) {
+          //  already create the collection and index on the slave
+          const c = db._create(cn);
+          c.ensureIndex(state.indexDef);
+        },
+        function (state) {
+          const c = db._collection(cn);
+          const i = c.index(state.indexDef.id);
+          assertEqual(state.indexProps.id, i.id);
```

Comment:
should we assert here that index names be identical too?
(in this test and the following)